src/query/config/src/global.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use common_base::base::Singleton;
+use common_exception::Result;
+use once_cell::sync::OnceCell;
+
+use crate::Config;
+
+pub struct GlobalConfig;
+
+static GLOBAL_CONFIG: OnceCell<Singleton<Arc<Config>>> = OnceCell::new();
+
+impl GlobalConfig {
+    pub fn init(config: Config, v: Singleton<Arc<Config>>) -> Result<()> {
```

Comment:
Can we add a unit test for the singleton?

Reply:
What behaviors do we need to test?

Reply:
An important test is to initialize multiple times and to ensure that the singleton is correct and works as well.